net/sshuttle/files/sshuttle.init

```diff
@@ -0,0 +1,108 @@
+#!/bin/sh /etc/rc.common
+#
+# Copyright (C) 2022 Thomas Cujé
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+# shellcheck disable=SC2034
+START=99
+STOP=01
+USE_PROCD=1
+
+_log() {
+  logger -p daemon.info -t sshuttle "$@"
+}
+
+append_params() {
+  for p in "$@"; do
+    eval "v=\$$p"
+    for v in $v; do
+      if [ -n "$v" ] && [ "$v" != "no" ]; then
+        if [ "$p" != "subnets" ]; then
+          p="--$( echo "$p" | sed -e 's|_|-|g' )"
+          if [ "$v" = "yes" ]; then
+            args="$args $p"
+          else
+            args="$args $p=$v"
+          fi
+        else
+          args="$args $v"
+        fi
+      fi
+    done
+  done
+
+  ARGS="${args# *}"
+}
+
+validate_gateway_section() {
+  uci_load_validate sshuttle gateway "$1" "$2" \
+    'listen:string' \
+    'subnets:list(string)' \
+    'method:or("auto", "nat", "nft", "tproxy", "pf", "ipfw")' \
+    'auto_hosts:or("yes", "no")' \
+    'auto_nets:or("yes", "no")' \
+    'dns:or("yes", "no")' \
+    'ns_hosts:list(host)' \
+    'to_ns:host' \
+    'python:string' \
+    'exclude_subnet:list(string)' \
+    'exclude_file:file' \
+    'verbose:or("yes", "no")' \
+    'ssh_cmd:string' \
+    'seed_hosts:list(host)' \
+    'latency_control:or("yes", "no")' \
+    'latency_buffer_size:integer' \
+    'subnets_file:file' \
+    'syslog:or("yes", "no")' \
+    'disable_ipv6:or("yes", "no")' \
+    'mark:string'
+}
+
+load_gateway() {
+  gateway="$1"
+
+  [ "$2" = 0 ] || { _err "gateway $gateway: validation failed"; return 1; }
+
+  append_params listen subnets method auto_hosts auto_nets dns ns_hosts to_ns \
+    python exclude_subnet exclude_file verbose ssh_cmd seed_hosts latency_control \
+    latency_buffer_size subnets_file syslog disable_ipv6 mark
+
+  ARGS="--remote=$1 --pidfile=/var/run/sshuttle.$1.pid $ARGS"
+  _log "Using sshuttle instance $gateway with args: $ARGS"
+
+  procd_open_instance "$gateway"
+  procd_set_param command "/usr/bin/sshuttle" $ARGS
+  procd_set_param stdout 1
+  procd_set_param stderr 1
+  procd_set_param respawn 0 60 1
+  procd_close_instance
+}
+
+service_triggers() {
+  procd_add_reload_trigger "sshuttle"
+
+  procd_open_validate
+  validate_gateway_section "$@"
+  procd_close_validate
+}
+
+start_service() {
+  _log "Starting sshuttle instances"
+  config_load "sshuttle"
+  config_foreach validate_gateway_section "gateway" load_gateway
+}
+
+stop_service() {
+  _log "Stopping sshuttle instances"
```

Comment:
I'm not sure if it is required. procd should handle this already, and `stop_service` is called even before procd kills the processes[^1].

[^1]: https://openwrt.org/docs/guide-developer/procd-init-scripts#stopping_services

Reply:
Thanks for the hint, I will look into this.

I'm not sure, how procd handles the (possibly) multiple processes for each connetion and terminates all of them correctly.

Reply:
Works like charm!

While checking I found that arguments from each instance are forwarded to the following ones. Fixed now, see my last commit.